### PR TITLE
Extract error message generation into a helper

### PR DESCRIPTION
**Pull request by stevenwilkin**
Going to need this for the Devise forms that will be pulled in soon

### CODE REVIEW

**Comment by stevenwilkin** [comment]
Updated with a helper/partial double punch per @gjvis's suggestion

File: app/views/shared/_errors.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<% if local_assigns[:model].errors.any? %>
+  <div class="alert alert-danger" role="alert">
+    <h2 class="alert__heading">There's a problem</h2>
+    <ul>
+      <% local_assigns[:model].errors.full_messages.each do |msg| %>
```
Comment (gjvis) [inline app/views/shared/_errors.html.erb]: This call and the one one line 1 will error out with an unuseful error message if model is nil.

File: app/views/shared/_errors.html.erb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+        <li><%= msg %></li>
+      <% end %>
+    </ul>
+  </div>
+<% end %>
```
Comment (gjvis) [inline app/views/shared/_errors.html.erb]: The partial actually only needs the errors, not the whole model (although that might be an implementation detail, your call)

File: app/views/shared/_errors.html.erb (hunk 1 above)
Comment (stevenwilkin) [inline app/views/shared/_errors.html.erb]: That did cross my mind. The helper that renders this partial should never be called if the model is nil but some defensive programming here could be in order

File: app/views/shared/_errors.html.erb (hunk 2 above)
Comment (stevenwilkin) [inline app/views/shared/_errors.html.erb]: I'm happy to have the model here considering how tightly coupled to `ActiveModel` the whole thing is

**Comment by stevenwilkin** [comment]
Updated with some defensive programming

File: app/views/shared/_errors.html.erb (hunk 2 above)
Comment (gjvis) [inline app/views/shared/_errors.html.erb]: :+1:

File: app/views/shared/_errors.html.erb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<% if local_assigns.fetch(:model, nil) && local_assigns[:model].errors.any? %>
```
Comment (gjvis) [inline app/views/shared/_errors.html.erb]: `local_assigns.include?(:model)`

Comment (stevenwilkin) [inline app/views/shared/_errors.html.erb]: Good call, I didn't know that existed, looks much neater

Comment (gjvis) [inline app/views/shared/_errors.html.erb]: I think local_assigns is just a `Hash` (although, knowing Rails, could be anything, hehe)

**Comment by stevenwilkin** [comment]
Build is passing, wanna hit the merge button?

**Comment by gjvis** [comment]
You go for it, I'm eyeballs deep in pub/sub. SO CLOSE TO A PR.

**Comment by gjvis** [comment]
:ship: :ship: :ship:

**Comment by stevenwilkin** [comment]
10 4 good buddy

**Pull request merged**